File: LeetCode/cpp/1021.cpp
class Solution {
public:
    string removeOuterParentheses(string S) {
        string ret;
        int empty = 0, start = 0;
        for (int i = 0; i < S.size(); ++i) {
            char c = S[i];
            if (c == '(')
                ++empty;
            else if (c == ')')
                --empty;
            if (empty == 0) {
                ret += S.substr(start + 1, i - start - 1);
                start = i + 1;
            }
        }
        return ret;
    }
};

File: LeetCode/python3/1021.py
class Solution:
    def removeOuterParentheses(self, S: str) -> str:
        empty, start, ret = 0, 0, ""
        for i in range(len(S)):
            if S[i] == '(':
                empty += 1
            elif S[i] == ')':
                empty -= 1
            if empty == 0:
                ret += S[start + 1:i]
                start = i + 1
        return ret
